File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/go/exercises/practice/word-search/word_search.go
package wordsearch

import "fmt"

func Solve(words []string, puzzle []string) (map[string][2][2]int, error) {
	result := make(map[string][2][2]int)
	directions := [][]int{
		{-1, 0},  // up
		{1, 0},   // down
		{0, -1},  // left
		{0, 1},   // right
		{-1, -1}, // up-left
		{-1, 1},  // up-right
		{1, -1},  // down-left
		{1, 1},   // down-right
	}

	for _, word := range words {
		if len(word) == 0 {
			continue
		}
		found := false
	wordLoop:
		for x, row := range puzzle {
			for y := range row {
				if row[y] != word[0] {
					continue
				}
				for _, dir := range directions {
					dx, dy := dir[0], dir[1]
					match := true
					for i := 0; i < len(word); i++ {
						nx := x + dx*i
						ny := y + dy*i
						if nx < 0 || nx >= len(puzzle) || ny < 0 || ny >= len(puzzle[nx]) {
							match = false
							break
						}
						if puzzle[nx][ny] != word[i] {
							match = false
							break
						}
					}
					if match {
						endX := x + dx*(len(word)-1)
						endY := y + dy*(len(word)-1)
						result[word] = [2][2]int{
							{y, x},
							{endY, endX},
						}
						found = true
						break wordLoop
					}
				}
			}
		}
		if !found {
			return nil, fmt.Errorf("word '%s' not found", word)
		}
	}
	return result, nil
}

File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/javascript/exercises/practice/word-search/word-search.js
//
// This is only a SKELETON file for the 'Word Search' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

class WordSearch {
  constructor(grid) {
    this.grid = grid.map(row => row.split(''));
    this.rows = this.grid.length;
    this.cols = this.rows > 0 ? this.grid[0].length : 0;
  }

  find(words) {
    const directions = [
      [0, 1],   // Right
      [1, 0],   // Down
      [0, -1],  // Left
      [-1, 0],  // Up
      [1, 1],   // Down-Right
      [1, -1],  // Down-Left
      [-1, 1],  // Up-Right
      [-1, -1]  // Up-Left
    ];
    
    const result = {};
    
    for (const word of words) {
      outer:
      for (let r = 0; r < this.rows; r++) {
        for (let c = 0; c < this.cols; c++) {
          if (this.grid[r][c] === word[0]) {
            for (const [dr, dc] of directions) {
              let found = true;
              let endRow = r, endCol = c;
              
              for (let i = 1; i < word.length; i++) {
                const nr = r + dr * i;
                const nc = c + dc * i;
                
                if (nr < 0 || nr >= this.rows || nc < 0 || nc >= this.cols || 
                    this.grid[nr][nc] !== word[i]) {
                  found = false;
                  break;
                }
                endRow = nr;
                endCol = nc;
              }
              
              if (found) {
                result[word] = {
                  start: [r + 1, c + 1],
                  end: [endRow + 1, endCol + 1]
                };
                break outer;
              }
            }
          }
        }
      }
    }
    
    return result;
  }
}

export default WordSearch;
